
syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.sensormanagement";
option java_outer_classname = "SensorManagement";
option objc_class_prefix = "HLW";

package sensormanagement;

// The data service definition
service SensorManagement {
    rpc GetAllSensorsOfType(AllSensorsOfTypeRequest) returns (AllSensorsOfTypeResponse) {}
    rpc GetAllSensorsOnVessel(AllSensorsOnVesselRequest) returns (AllSensorsOnVesselResponse) {}
}

message AllSensorsOfTypeRequest {
    SensorType type = 1;
}

message AllSensorsOfTypeResponse {
    repeated Sensor sensors = 1;
}

message AllSensorsOnVesselRequest {
    string vesselID = 1;
}

message AllSensorsOnVesselResponse {
    repeated Sensor sensors = 1;
}

enum SensorType {
    UNKNOWN = 0;
    OPTICAL = 1;
    INFRARED = 2;
    RADAR = 3;
    LIDAR = 4;
}

message Sensor {
    SensorType type = 1;
    int32 sensorWidth = 2;
    int32 sensorHeight = 3;
    string ipAddress = 4;
    int32 port = 5;
}
